app/src/main/java/ch/epfl/sdp/drone3d/map/MapboxUserDrawer.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.map
+
+import android.graphics.Color
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.plugins.annotation.Circle
+import com.mapbox.mapboxsdk.plugins.annotation.CircleManager
+import com.mapbox.mapboxsdk.plugins.annotation.CircleOptions
+import com.mapbox.mapboxsdk.utils.ColorUtils
+
+/**
+ * This class draw the location of the home of the drone on the map
+ *
+ * This class is taken from the Fly2Find project with one adaptation :
+ * - the SymbolManager was replaced by a CircleManager, and the modifications implied by this change
+ */
+class MapboxUserDrawer(mapView: MapView, mapboxMap: MapboxMap, style: Style) : MapboxDrawer {
+    private var circleManager = CircleManager(mapView, mapboxMap, style)
+    private lateinit var marker: Circle
+    private var reset: Boolean = false
+
+    /**
+     * Draws the [location] of the home on the map
+     */
+    fun showUser(location: LatLng?) {
+        if (location == null) {
+            circleManager.deleteAll()
+            reset = true
+        } else if (!::marker.isInitialized || reset) {
```

Comment:
Couldn't you use only the reset ? Something like setting it to true by default any only using it as condition

Reply:
```suggestion
        } else if (reset) {
```

Reply:
Yes, I changed it here and in other similar classes